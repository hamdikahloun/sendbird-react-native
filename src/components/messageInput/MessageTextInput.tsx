import React, {useState} from 'react';
import {StyleSheet, TextInput} from 'react-native';

type MessageTextInputProps = {
  onChangeMessage?: (message: string) => void;
  messageTextInputRef?: React.MutableRefObject<TextInput | undefined>;
  enabled?: boolean;
};

/**
 * MessageTextInput - This UI component displays Message TextInput
 *
 * @example
 *
 * <MessageTextInput messageTextInputRef={messageTextInputRef} onMessageChange={onMessageChange} enabled={enabled}/>
 */
export const MessageTextInput = ({
  onChangeMessage,
  messageTextInputRef,
  enabled,
}: MessageTextInputProps) => {
  const [message, setMessage] = useState('');

  const onChangeText = (text: string) => {
    setMessage(text);
    onChangeMessage?.(text);
  };

  return (
    <TextInput
      ref={messageTextInputRef as React.LegacyRef<TextInput>}
      testID="message"
      style={styles.input}
      onChangeText={onChangeText}
      value={message}
      placeholder="Aa"
      editable={enabled}
    />
  );
};

const styles = StyleSheet.create({
  input: {
    height: 50,
    marginEnd: 8,
    borderWidth: 1,
    borderColor: '#F9F9F9',
    padding: 10,
    backgroundColor: '#EFEFEF',
    borderRadius: 10,
    flex: 1,
  },
});
