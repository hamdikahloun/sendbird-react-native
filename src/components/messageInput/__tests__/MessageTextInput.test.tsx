import React from 'react';
import {render} from '@/../testing';
import {MessageTextInput} from '../MessageTextInput';

describe('MessageTextInput', () => {
  it('should render correctly', async () => {
    const screen = render(<MessageTextInput />).toJSON();
    expect(screen).toMatchSnapshot();
  });
});
