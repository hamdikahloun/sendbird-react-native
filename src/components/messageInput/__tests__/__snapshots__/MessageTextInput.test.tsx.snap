// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageTextInput should render correctly 1`] = `
<TextInput
  onChangeText={[Function]}
  placeholder="Aa"
  style={
    Object {
      "backgroundColor": "#EFEFEF",
      "borderColor": "#F9F9F9",
      "borderRadius": 10,
      "borderWidth": 1,
      "flex": 1,
      "height": 50,
      "marginEnd": 8,
      "padding": 10,
    }
  }
  testID="message"
  value=""
/>
`;
